subtrack.MAUI/Services/SubscriptionsCalculator.cs

```diff
@@ -65,14 +65,26 @@ public DateTime GetNextPaymentDate(Subscription subscription)
     {
         int startDay = subscription.FirstPaymentDay;
         var lastPayment = subscription.LastPayment;
-
-        var nextMonthDate = lastPayment.AddMonths(1);
-        var nextMonthTotalDays = DateTime.DaysInMonth(nextMonthDate.Year, nextMonthDate.Month);
-
-        if (startDay > nextMonthTotalDays)
-            return nextMonthDate;
-
-        return new DateTime(nextMonthDate.Year, nextMonthDate.Month, startDay);
+        switch (subscription.BillingOccurrence)
+        {
+            case BillingOccurrence.Week:
+                var nextWeekDate = lastPayment.AddDays(7*subscription.BillingInterval);
+                return new DateTime(nextWeekDate.Year, nextWeekDate.Month, nextWeekDate.Day);
+            case BillingOccurrence.Month:
+                var nextMonthDate = lastPayment.AddMonths(subscription.BillingInterval);
+                var nextMonthTotalDays = DateTime.DaysInMonth(nextMonthDate.Year, nextMonthDate.Month);
+
+                if (startDay > nextMonthTotalDays)
+                    return nextMonthDate;
+
+                return new DateTime(nextMonthDate.Year, nextMonthDate.Month, startDay);
+            case BillingOccurrence.Year:
+                var nextYearDate = lastPayment.AddYears(subscription.BillingInterval);
```

Comment:
Dont forget about startday here aswell

Reply:
Forgot about that, but seems like AddYears takes into account leap years e.g. 
https://learn.microsoft.com/en-us/dotnet/api/system.datetime.addyears?view=net-7.0
If value + DateTime.Year is not a leap year, the return value represents the day before the leap day in that year. For example, if one year is added to February 29, 2012, the date returned is February 28, 2013.

Reply:
Yes it does account for leap years, just like .addmonths accounts for the last day of the month but that doesnt change the fact that we need to account for the start day. It only considers last payment not start payment afaik

Example: feb 29 start day, add one year its 28, then when its leap year again its 28 like the previous year instead of 29 because we did not account for the starting day